kie-server-parent/kie-server-remote/kie-server-client/src/main/java/org/kie/server/client/CaseServicesClient.java

```diff
@@ -0,0 +1,117 @@
+
+import org.kie.server.api.model.cases.CaseAdHocFragment;
+import org.kie.server.api.model.cases.CaseComment;
+import org.kie.server.api.model.cases.CaseDefinition;
+import org.kie.server.api.model.cases.CaseFile;
+import org.kie.server.api.model.cases.CaseInstance;
+import org.kie.server.api.model.cases.CaseMilestone;
+import org.kie.server.api.model.cases.CaseRoleAssignment;
+import org.kie.server.api.model.cases.CaseStage;
+import org.kie.server.api.model.instance.NodeInstance;
+import org.kie.server.api.model.instance.ProcessInstance;
+
+public interface CaseServicesClient {
+
+    String startCase(String containerId, String caseDefinitionId);
+
+    String startCase(String containerId, String caseDefinitionId, CaseFile caseFile);
+
+    CaseInstance getCaseInstance(String containerId, String caseId);
+
+    CaseInstance getCaseInstance(String containerId, String caseId, boolean withData, boolean withRoles, boolean withMilestones, boolean withStages);
+
+    void cancelCaseInstance(String containerId, String caseId);
+
+    void destroyCaseInstance(String containerId, String caseId);
+
+    Map<String, Object> getCaseInstanceData(String containerId, String caseId);
+
+    Object getCaseInstanceData(String containerId, String caseId, String name);
+
+    void putCaseInstanceData(String containerId, String caseId, Map<String, Object> data);
+
+    void putCaseInstanceData(String containerId, String caseId, String name, Object data);
+
+    void removeCaseInstanceData(String containerId, String caseId, String... names);
+
+    void addDynamicUserTask(String containerId, String caseId, String name, String description, String actors, String groups, Map<String, Object> data);
+
+    void addDynamicTask(String containerId, String caseId, String nodeType, String name, Map<String, Object> data);
+
+    void addDynamicUserTaskToStage(String containerId, String caseId, String stageId, String name, String description, String actors, String groups, Map<String, Object> data);
+
+    void addDynamicTaskToStage(String containerId, String caseId, String stageId, String nodeType, String name, Map<String, Object> data);
+
+    void addDynamicSubProcess(String containerId, String caseId, String processId, Map<String, Object> data);
+
+    void addDynamicSubProcessToStage(String containerId, String caseId, String stageId,  String processId, Map<String, Object> data);
+
+    void triggerAdHocFragment(String containerId, String caseId, String adHocName, Map<String, Object> data);
+
+    void triggerAdHocFragmentInStage(String containerId, String caseId, String stageId, String adHocName, Map<String, Object> data);
+
+    List<CaseMilestone> getMilestones(String containerId, String caseId, boolean achievedOnly, Integer page, Integer pageSize);
+
+    List<CaseStage> getStages(String containerId, String caseId, boolean activeOnly, Integer page, Integer pageSize);
+
+    List<CaseAdHocFragment> getAdHocFragments(String containerId, String caseId);
```

Comment:
Would this method consider the stage that the case is in? For instance, certain AdHoc fragment would be only available inside a stage and another fragment is available at any stage in the case.